src/librustc/ty/sty.rs

```diff
@@ -209,6 +209,19 @@ pub enum TyKind<'tcx> {
     Error,
 }
 
+impl<'tcx> TyKind<'tcx> {
+    pub fn suggestion(&self) -> Option<&'static str> {
```

Comment:
I moved this here because it'd now be used in multiple places so it felt like the natural place to be. Where would you recommend this go (given that it is moving code from typeck, and using it in rustc/ty and typeck).

Reply:
I would put it in rustc_typeck; the only direct usage in rustc goes away if you move the method above as well.